src/Catty/ViewController/Stage/StagePresenterViewControllerResourcesExtension.swift

```diff
@@ -31,11 +31,14 @@ import CoreBluetooth
         self.showLoadingView()
 
         DispatchQueue.global(qos: .userInitiated).async {
-            self.project = Project.init(loadingInfo: Util.lastUsedProjectLoadingInfo())!
-            self.formulaManager = FormulaManager(stageSize: Util.screenSize(true), landscapeMode: self.project.header.landscapeMode)
-            let readyToStart = self.notifyUserAboutUnavailableResources(navigationController: navigationController)
-
+            guard let project = Project.init(loadingInfo: Util.lastUsedProjectLoadingInfo()) else {
+                NSLog(kLocalizedInvalidZip)
```

Comment:
Instead of printing a log message, an alert should be displayed to the user. Also do not forget to hide the `loadingView` again.